docs/data-sources/cluster_node_pool.md

```diff
@@ -91,6 +91,22 @@ Required:
 - `class` (String) Nodepool instance type
 - `storage_class` (String) Storage Class to be used for storage of the disks which store the root filesystem of the nodes
 
+Optional:
+
+- `failure_domain` (String) Configure the failure domain of node pool. The potential values could be found using cluster:options api. This parameter will be ignored by the backend if the TKG service vsphere cluster doesn't support.
```

Comment:
One open comment from @vmw-vjn : https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/236#discussion_r1296769353

Reply:
Ignore the above comment and I will take up the node pool resource doc update separately in a [PR](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/236). Split this into 2 PR's as suggested.

Reply:
Done